hadoop-ozone/dist/src/main/smoketest/basic/links.robot

```diff
@@ -0,0 +1,152 @@
+                        Should Contain              ${source_list}    key2
+
+Key delete passthrough
+                        Execute                     ozone sh key delete ${target}/link1/key2
+    ${source_list} =    Execute                     ozone sh key list ${source}/bucket1 | jq -r '.name'
+                        Should Not Contain          ${source_list}    key2
+
+Bucket list contains links
+    ${result} =         Execute                     ozone sh bucket list ${target}
+                        Should Contain              ${result}         link1
+                        Should Contain              ${result}         dangling-link
+
+Bucket info shows source
+    ${result} =         Execute                     ozone sh bucket info ${target}/link1 | jq -r '.sourceVolume, .sourceBucket' | xargs
+                        Should Be Equal             ${result}    ${source} bucket1
+
+Source and target have separate ACLs
+    Execute       ozone sh bucket addacl --acl user:user1:rwxy ${target}/link1
+    Verify ACL    bucket    ${target}/link1      USER    user1    READ WRITE READ_ACL WRITE_ACL
+    Verify ACL    bucket    ${source}/bucket1    USER    user1    ${EMPTY}
+
+    Execute       ozone sh bucket addacl --acl group:group2:r ${source}/bucket1
+    Verify ACL    bucket    ${target}/link1      GROUP   group2    ${EMPTY}
+    Verify ACL    bucket    ${source}/bucket1    GROUP   group2    READ
+
+Buckets and links share namespace
+                        Execute                     ozone sh bucket link ${source}/bucket2 ${target}/link2
+    ${result} =         Execute And Ignore Error    ozone sh bucket create ${target}/link2
+                        Should Contain              ${result}    BUCKET_ALREADY_EXISTS
+
+                        Execute                     ozone sh bucket create ${target}/bucket3
+    ${result} =         Execute And Ignore Error    ozone sh bucket link ${source}/bucket1 ${target}/bucket3
+                        Should Contain              ${result}    BUCKET_ALREADY_EXISTS
+
+Can follow link with read access
+    Run Keyword if    '${SECURITY_ENABLED}' == 'true'    Can follow link with read access
+
+Cannot follow link without read access
+    Run Keyword if    '${SECURITY_ENABLED}' == 'true'    Cannot follow link without read access
+
+ACL verified on source bucket
+    Run Keyword if    '${SECURITY_ENABLED}' == 'true'    ACL verified on source bucket
+
+Loop in link chain is detected
+                        Execute                     ozone sh bucket link ${target}/loop1 ${target}/loop2
+                        Execute                     ozone sh bucket link ${target}/loop2 ${target}/loop3
+                        Execute                     ozone sh bucket link ${target}/loop3 ${target}/loop1
+    ${result} =         Execute And Ignore Error    ozone sh key list ${target}/loop2
+                        Should Contain              ${result}    INTERNAL_ERROR
```

Comment:
This needs to be updated, once error code is modified.

Reply:
Done.